<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Marnin Wolfe" />


<title>Check prediction accuracy</title>

<script src="site_libs/header-attrs-2.7/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/yeti.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
<script src="site_libs/pagedtable-1.1/js/pagedtable.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>


<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    background-color: #232629;
    color: #7a7c7d;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #7a7c7d;  padding-left: 4px; }
div.sourceCode
  { color: #cfcfc2; background-color: #232629; }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span { color: #cfcfc2; } /* Normal */
code span.al { color: #95da4c; background-color: #4d1f24; font-weight: bold; } /* Alert */
code span.an { color: #3f8058; } /* Annotation */
code span.at { color: #2980b9; } /* Attribute */
code span.bn { color: #f67400; } /* BaseN */
code span.bu { color: #7f8c8d; } /* BuiltIn */
code span.cf { color: #fdbc4b; font-weight: bold; } /* ControlFlow */
code span.ch { color: #3daee9; } /* Char */
code span.cn { color: #27aeae; font-weight: bold; } /* Constant */
code span.co { color: #7a7c7d; } /* Comment */
code span.cv { color: #7f8c8d; } /* CommentVar */
code span.do { color: #a43340; } /* Documentation */
code span.dt { color: #2980b9; } /* DataType */
code span.dv { color: #f67400; } /* DecVal */
code span.er { color: #da4453; text-decoration: underline; } /* Error */
code span.ex { color: #0099ff; font-weight: bold; } /* Extension */
code span.fl { color: #f67400; } /* Float */
code span.fu { color: #8e44ad; } /* Function */
code span.im { color: #27ae60; } /* Import */
code span.in { color: #c45b00; } /* Information */
code span.kw { color: #cfcfc2; font-weight: bold; } /* Keyword */
code span.op { color: #cfcfc2; } /* Operator */
code span.ot { color: #27ae60; } /* Other */
code span.pp { color: #27ae60; } /* Preprocessor */
code span.re { color: #2980b9; background-color: #153042; } /* RegionMarker */
code span.sc { color: #3daee9; } /* SpecialChar */
code span.ss { color: #da4453; } /* SpecialString */
code span.st { color: #f44f4f; } /* String */
code span.va { color: #27aeae; } /* Variable */
code span.vs { color: #da4453; } /* VerbatimString */
code span.wa { color: #da4453; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>







<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-sm-12 col-md-4 col-lg-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-sm-12 col-md-8 col-lg-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">NRCRI_2021GS</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Imputation (DCas21_5841)
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="convertDCas21_5841_ToVCF.html">Conversion DArT report DCas21_5841 to VCF</a>
    </li>
    <li>
      <a href="ImputeDCas21_5841.html">Impute DCas21_5841</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Prediction (April 2021)
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="01-cleanTPdata.html">Data Cleaning</a>
    </li>
    <li>
      <a href="02-GetBLUPs.html">Get BLUPs</a>
    </li>
    <li>
      <a href="03-CrossValidation.html">Check prediction accuracy</a>
    </li>
    <li>
      <a href="04-GetGBLUPs.html">Get GEBV/GETGV for GS C3b</a>
    </li>
  </ul>
</li>
<li>
  <a href="05-Results.html">Results</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Other
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="about.html">About</a>
    </li>
    <li>
      <a href="license.html">License</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/wolfemd/NRCRI_2021GS">
    <span class="fas fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Check prediction accuracy</h1>
<h4 class="author">Marnin Wolfe</h4>
<h4 class="date">2021-May-03</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-05-05
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>NRCRI_2021GS/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20210427code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20210427)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20210427code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20210427)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomwolfemdNRCRI2021GStree94ec81122c0b7a209fbf75bf47ca3e2101a0f529targetblank94ec811a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/wolfemd/NRCRI_2021GS/tree/94ec81122c0b7a209fbf75bf47ca3e2101a0f529" target="_blank">94ec811</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomwolfemdNRCRI2021GStree94ec81122c0b7a209fbf75bf47ca3e2101a0f529targetblank94ec811a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/wolfemd/NRCRI_2021GS/tree/94ec81122c0b7a209fbf75bf47ca3e2101a0f529" target="_blank">94ec811</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    data/.DS_Store

Untracked files:
    Untracked:  code/convertDart2vcf.R
    Untracked:  code/gsFunctions.R
    Untracked:  code/imputationFunctions.R
    Untracked:  data/DatabaseDownload_2021April29/
    Untracked:  data/GBSdataMasterList_31818.csv
    Untracked:  data/IITA_GBStoPhenoMaster_33018.csv
    Untracked:  data/NRCRI_GBStoPhenoMaster_40318.csv
    Untracked:  data/Report-DCas21-5841/
    Untracked:  data/chr1_RefPanelAndGSprogeny_ReadyForGP_72719.fam
    Untracked:  output/BeagleLogs/
    Untracked:  output/DosageMatrix_DCas21_5841_WA_REFimputedAndFiltered.rds
    Untracked:  output/DosageMatrix_NRCRI_2021May03.rds
    Untracked:  output/GEBV_NRCRI_ModelA_2021May03.csv
    Untracked:  output/GETGV_NRCRI_ModelADE_2021May03.csv
    Untracked:  output/Kinship_AD_NRCRI_2021May03.rds
    Untracked:  output/Kinship_A_NRCRI_2021May03.rds
    Untracked:  output/Kinship_D_NRCRI_2021May03.rds
    Untracked:  output/NRCRI_CleanedTrialData_2021May03.rds
    Untracked:  output/NRCRI_ExptDesignsDetected_2021May03.rds
    Untracked:  output/NRCRI_blupsForModelTraining_twostage_asreml_2021May03.rds
    Untracked:  output/NRCRI_trials_NOT_identifiable.csv
    Untracked:  output/OnlyChosen_germplasmName_to_FullSampleName_matches_NRCRI_2021May03.csv
    Untracked:  output/chr10_DCas21_5841_WA_REFimputed.INFO
    Untracked:  output/chr10_DCas21_5841_WA_REFimputed.hwe
    Untracked:  output/chr10_DCas21_5841_WA_REFimputed.log
    Untracked:  output/chr10_DCas21_5841_WA_REFimputed.sitesPassing
    Untracked:  output/chr10_DCas21_5841_WA_REFimputed.vcf.gz
    Untracked:  output/chr10_DCas21_5841_WA_REFimputedAndFiltered.alleleToCount
    Untracked:  output/chr10_DCas21_5841_WA_REFimputedAndFiltered.bed
    Untracked:  output/chr10_DCas21_5841_WA_REFimputedAndFiltered.bim
    Untracked:  output/chr10_DCas21_5841_WA_REFimputedAndFiltered.fam
    Untracked:  output/chr10_DCas21_5841_WA_REFimputedAndFiltered.log
    Untracked:  output/chr10_DCas21_5841_WA_REFimputedAndFiltered.nosex
    Untracked:  output/chr10_DCas21_5841_WA_REFimputedAndFiltered.raw
    Untracked:  output/chr10_DCas21_5841_WA_REFimputedAndFiltered.sitesWithAlleles
    Untracked:  output/chr10_DCas21_5841_WA_REFimputedAndFiltered.vcf.gz
    Untracked:  output/chr11_DCas21_5841_WA_REFimputed.INFO
    Untracked:  output/chr11_DCas21_5841_WA_REFimputed.hwe
    Untracked:  output/chr11_DCas21_5841_WA_REFimputed.log
    Untracked:  output/chr11_DCas21_5841_WA_REFimputed.sitesPassing
    Untracked:  output/chr11_DCas21_5841_WA_REFimputed.vcf.gz
    Untracked:  output/chr11_DCas21_5841_WA_REFimputedAndFiltered.alleleToCount
    Untracked:  output/chr11_DCas21_5841_WA_REFimputedAndFiltered.bed
    Untracked:  output/chr11_DCas21_5841_WA_REFimputedAndFiltered.bim
    Untracked:  output/chr11_DCas21_5841_WA_REFimputedAndFiltered.fam
    Untracked:  output/chr11_DCas21_5841_WA_REFimputedAndFiltered.log
    Untracked:  output/chr11_DCas21_5841_WA_REFimputedAndFiltered.nosex
    Untracked:  output/chr11_DCas21_5841_WA_REFimputedAndFiltered.raw
    Untracked:  output/chr11_DCas21_5841_WA_REFimputedAndFiltered.sitesWithAlleles
    Untracked:  output/chr11_DCas21_5841_WA_REFimputedAndFiltered.vcf.gz
    Untracked:  output/chr12_DCas21_5841_WA_REFimputed.INFO
    Untracked:  output/chr12_DCas21_5841_WA_REFimputed.hwe
    Untracked:  output/chr12_DCas21_5841_WA_REFimputed.log
    Untracked:  output/chr12_DCas21_5841_WA_REFimputed.sitesPassing
    Untracked:  output/chr12_DCas21_5841_WA_REFimputed.vcf.gz
    Untracked:  output/chr12_DCas21_5841_WA_REFimputedAndFiltered.alleleToCount
    Untracked:  output/chr12_DCas21_5841_WA_REFimputedAndFiltered.bed
    Untracked:  output/chr12_DCas21_5841_WA_REFimputedAndFiltered.bim
    Untracked:  output/chr12_DCas21_5841_WA_REFimputedAndFiltered.fam
    Untracked:  output/chr12_DCas21_5841_WA_REFimputedAndFiltered.log
    Untracked:  output/chr12_DCas21_5841_WA_REFimputedAndFiltered.nosex
    Untracked:  output/chr12_DCas21_5841_WA_REFimputedAndFiltered.raw
    Untracked:  output/chr12_DCas21_5841_WA_REFimputedAndFiltered.sitesWithAlleles
    Untracked:  output/chr12_DCas21_5841_WA_REFimputedAndFiltered.vcf.gz
    Untracked:  output/chr13_DCas21_5841_WA_REFimputed.INFO
    Untracked:  output/chr13_DCas21_5841_WA_REFimputed.hwe
    Untracked:  output/chr13_DCas21_5841_WA_REFimputed.log
    Untracked:  output/chr13_DCas21_5841_WA_REFimputed.sitesPassing
    Untracked:  output/chr13_DCas21_5841_WA_REFimputed.vcf.gz
    Untracked:  output/chr13_DCas21_5841_WA_REFimputedAndFiltered.alleleToCount
    Untracked:  output/chr13_DCas21_5841_WA_REFimputedAndFiltered.bed
    Untracked:  output/chr13_DCas21_5841_WA_REFimputedAndFiltered.bim
    Untracked:  output/chr13_DCas21_5841_WA_REFimputedAndFiltered.fam
    Untracked:  output/chr13_DCas21_5841_WA_REFimputedAndFiltered.log
    Untracked:  output/chr13_DCas21_5841_WA_REFimputedAndFiltered.nosex
    Untracked:  output/chr13_DCas21_5841_WA_REFimputedAndFiltered.raw
    Untracked:  output/chr13_DCas21_5841_WA_REFimputedAndFiltered.sitesWithAlleles
    Untracked:  output/chr13_DCas21_5841_WA_REFimputedAndFiltered.vcf.gz
    Untracked:  output/chr14_DCas21_5841_WA_REFimputed.INFO
    Untracked:  output/chr14_DCas21_5841_WA_REFimputed.hwe
    Untracked:  output/chr14_DCas21_5841_WA_REFimputed.log
    Untracked:  output/chr14_DCas21_5841_WA_REFimputed.sitesPassing
    Untracked:  output/chr14_DCas21_5841_WA_REFimputed.vcf.gz
    Untracked:  output/chr14_DCas21_5841_WA_REFimputedAndFiltered.alleleToCount
    Untracked:  output/chr14_DCas21_5841_WA_REFimputedAndFiltered.bed
    Untracked:  output/chr14_DCas21_5841_WA_REFimputedAndFiltered.bim
    Untracked:  output/chr14_DCas21_5841_WA_REFimputedAndFiltered.fam
    Untracked:  output/chr14_DCas21_5841_WA_REFimputedAndFiltered.log
    Untracked:  output/chr14_DCas21_5841_WA_REFimputedAndFiltered.nosex
    Untracked:  output/chr14_DCas21_5841_WA_REFimputedAndFiltered.raw
    Untracked:  output/chr14_DCas21_5841_WA_REFimputedAndFiltered.sitesWithAlleles
    Untracked:  output/chr14_DCas21_5841_WA_REFimputedAndFiltered.vcf.gz
    Untracked:  output/chr15_DCas21_5841_WA_REFimputed.INFO
    Untracked:  output/chr15_DCas21_5841_WA_REFimputed.hwe
    Untracked:  output/chr15_DCas21_5841_WA_REFimputed.log
    Untracked:  output/chr15_DCas21_5841_WA_REFimputed.sitesPassing
    Untracked:  output/chr15_DCas21_5841_WA_REFimputed.vcf.gz
    Untracked:  output/chr15_DCas21_5841_WA_REFimputedAndFiltered.alleleToCount
    Untracked:  output/chr15_DCas21_5841_WA_REFimputedAndFiltered.bed
    Untracked:  output/chr15_DCas21_5841_WA_REFimputedAndFiltered.bim
    Untracked:  output/chr15_DCas21_5841_WA_REFimputedAndFiltered.fam
    Untracked:  output/chr15_DCas21_5841_WA_REFimputedAndFiltered.log
    Untracked:  output/chr15_DCas21_5841_WA_REFimputedAndFiltered.nosex
    Untracked:  output/chr15_DCas21_5841_WA_REFimputedAndFiltered.raw
    Untracked:  output/chr15_DCas21_5841_WA_REFimputedAndFiltered.sitesWithAlleles
    Untracked:  output/chr15_DCas21_5841_WA_REFimputedAndFiltered.vcf.gz
    Untracked:  output/chr16_DCas21_5841_WA_REFimputed.INFO
    Untracked:  output/chr16_DCas21_5841_WA_REFimputed.hwe
    Untracked:  output/chr16_DCas21_5841_WA_REFimputed.log
    Untracked:  output/chr16_DCas21_5841_WA_REFimputed.sitesPassing
    Untracked:  output/chr16_DCas21_5841_WA_REFimputed.vcf.gz
    Untracked:  output/chr16_DCas21_5841_WA_REFimputedAndFiltered.alleleToCount
    Untracked:  output/chr16_DCas21_5841_WA_REFimputedAndFiltered.bed
    Untracked:  output/chr16_DCas21_5841_WA_REFimputedAndFiltered.bim
    Untracked:  output/chr16_DCas21_5841_WA_REFimputedAndFiltered.fam
    Untracked:  output/chr16_DCas21_5841_WA_REFimputedAndFiltered.log
    Untracked:  output/chr16_DCas21_5841_WA_REFimputedAndFiltered.nosex
    Untracked:  output/chr16_DCas21_5841_WA_REFimputedAndFiltered.raw
    Untracked:  output/chr16_DCas21_5841_WA_REFimputedAndFiltered.sitesWithAlleles
    Untracked:  output/chr16_DCas21_5841_WA_REFimputedAndFiltered.vcf.gz
    Untracked:  output/chr17_DCas21_5841_WA_REFimputed.INFO
    Untracked:  output/chr17_DCas21_5841_WA_REFimputed.hwe
    Untracked:  output/chr17_DCas21_5841_WA_REFimputed.log
    Untracked:  output/chr17_DCas21_5841_WA_REFimputed.sitesPassing
    Untracked:  output/chr17_DCas21_5841_WA_REFimputed.vcf.gz
    Untracked:  output/chr17_DCas21_5841_WA_REFimputedAndFiltered.alleleToCount
    Untracked:  output/chr17_DCas21_5841_WA_REFimputedAndFiltered.bed
    Untracked:  output/chr17_DCas21_5841_WA_REFimputedAndFiltered.bim
    Untracked:  output/chr17_DCas21_5841_WA_REFimputedAndFiltered.fam
    Untracked:  output/chr17_DCas21_5841_WA_REFimputedAndFiltered.log
    Untracked:  output/chr17_DCas21_5841_WA_REFimputedAndFiltered.nosex
    Untracked:  output/chr17_DCas21_5841_WA_REFimputedAndFiltered.raw
    Untracked:  output/chr17_DCas21_5841_WA_REFimputedAndFiltered.sitesWithAlleles
    Untracked:  output/chr17_DCas21_5841_WA_REFimputedAndFiltered.vcf.gz
    Untracked:  output/chr18_DCas21_5841_WA_REFimputed.INFO
    Untracked:  output/chr18_DCas21_5841_WA_REFimputed.hwe
    Untracked:  output/chr18_DCas21_5841_WA_REFimputed.log
    Untracked:  output/chr18_DCas21_5841_WA_REFimputed.sitesPassing
    Untracked:  output/chr18_DCas21_5841_WA_REFimputed.vcf.gz
    Untracked:  output/chr18_DCas21_5841_WA_REFimputedAndFiltered.alleleToCount
    Untracked:  output/chr18_DCas21_5841_WA_REFimputedAndFiltered.bed
    Untracked:  output/chr18_DCas21_5841_WA_REFimputedAndFiltered.bim
    Untracked:  output/chr18_DCas21_5841_WA_REFimputedAndFiltered.fam
    Untracked:  output/chr18_DCas21_5841_WA_REFimputedAndFiltered.log
    Untracked:  output/chr18_DCas21_5841_WA_REFimputedAndFiltered.nosex
    Untracked:  output/chr18_DCas21_5841_WA_REFimputedAndFiltered.raw
    Untracked:  output/chr18_DCas21_5841_WA_REFimputedAndFiltered.sitesWithAlleles
    Untracked:  output/chr18_DCas21_5841_WA_REFimputedAndFiltered.vcf.gz
    Untracked:  output/chr1_DCas21_5841_WA_REFimputed.INFO
    Untracked:  output/chr1_DCas21_5841_WA_REFimputed.hwe
    Untracked:  output/chr1_DCas21_5841_WA_REFimputed.log
    Untracked:  output/chr1_DCas21_5841_WA_REFimputed.sitesPassing
    Untracked:  output/chr1_DCas21_5841_WA_REFimputed.vcf.gz
    Untracked:  output/chr1_DCas21_5841_WA_REFimputedAndFiltered.alleleToCount
    Untracked:  output/chr1_DCas21_5841_WA_REFimputedAndFiltered.bed
    Untracked:  output/chr1_DCas21_5841_WA_REFimputedAndFiltered.bim
    Untracked:  output/chr1_DCas21_5841_WA_REFimputedAndFiltered.fam
    Untracked:  output/chr1_DCas21_5841_WA_REFimputedAndFiltered.log
    Untracked:  output/chr1_DCas21_5841_WA_REFimputedAndFiltered.nosex
    Untracked:  output/chr1_DCas21_5841_WA_REFimputedAndFiltered.raw
    Untracked:  output/chr1_DCas21_5841_WA_REFimputedAndFiltered.sitesWithAlleles
    Untracked:  output/chr1_DCas21_5841_WA_REFimputedAndFiltered.vcf.gz
    Untracked:  output/chr2_DCas21_5841_WA_REFimputed.INFO
    Untracked:  output/chr2_DCas21_5841_WA_REFimputed.hwe
    Untracked:  output/chr2_DCas21_5841_WA_REFimputed.log
    Untracked:  output/chr2_DCas21_5841_WA_REFimputed.sitesPassing
    Untracked:  output/chr2_DCas21_5841_WA_REFimputed.vcf.gz
    Untracked:  output/chr2_DCas21_5841_WA_REFimputedAndFiltered.alleleToCount
    Untracked:  output/chr2_DCas21_5841_WA_REFimputedAndFiltered.bed
    Untracked:  output/chr2_DCas21_5841_WA_REFimputedAndFiltered.bim
    Untracked:  output/chr2_DCas21_5841_WA_REFimputedAndFiltered.fam
    Untracked:  output/chr2_DCas21_5841_WA_REFimputedAndFiltered.log
    Untracked:  output/chr2_DCas21_5841_WA_REFimputedAndFiltered.nosex
    Untracked:  output/chr2_DCas21_5841_WA_REFimputedAndFiltered.raw
    Untracked:  output/chr2_DCas21_5841_WA_REFimputedAndFiltered.sitesWithAlleles
    Untracked:  output/chr2_DCas21_5841_WA_REFimputedAndFiltered.vcf.gz
    Untracked:  output/chr3_DCas21_5841_WA_REFimputed.INFO
    Untracked:  output/chr3_DCas21_5841_WA_REFimputed.hwe
    Untracked:  output/chr3_DCas21_5841_WA_REFimputed.log
    Untracked:  output/chr3_DCas21_5841_WA_REFimputed.sitesPassing
    Untracked:  output/chr3_DCas21_5841_WA_REFimputed.vcf.gz
    Untracked:  output/chr3_DCas21_5841_WA_REFimputedAndFiltered.alleleToCount
    Untracked:  output/chr3_DCas21_5841_WA_REFimputedAndFiltered.bed
    Untracked:  output/chr3_DCas21_5841_WA_REFimputedAndFiltered.bim
    Untracked:  output/chr3_DCas21_5841_WA_REFimputedAndFiltered.fam
    Untracked:  output/chr3_DCas21_5841_WA_REFimputedAndFiltered.log
    Untracked:  output/chr3_DCas21_5841_WA_REFimputedAndFiltered.nosex
    Untracked:  output/chr3_DCas21_5841_WA_REFimputedAndFiltered.raw
    Untracked:  output/chr3_DCas21_5841_WA_REFimputedAndFiltered.sitesWithAlleles
    Untracked:  output/chr3_DCas21_5841_WA_REFimputedAndFiltered.vcf.gz
    Untracked:  output/chr4_DCas21_5841_WA_REFimputed.INFO
    Untracked:  output/chr4_DCas21_5841_WA_REFimputed.hwe
    Untracked:  output/chr4_DCas21_5841_WA_REFimputed.log
    Untracked:  output/chr4_DCas21_5841_WA_REFimputed.sitesPassing
    Untracked:  output/chr4_DCas21_5841_WA_REFimputed.vcf.gz
    Untracked:  output/chr4_DCas21_5841_WA_REFimputedAndFiltered.alleleToCount
    Untracked:  output/chr4_DCas21_5841_WA_REFimputedAndFiltered.bed
    Untracked:  output/chr4_DCas21_5841_WA_REFimputedAndFiltered.bim
    Untracked:  output/chr4_DCas21_5841_WA_REFimputedAndFiltered.fam
    Untracked:  output/chr4_DCas21_5841_WA_REFimputedAndFiltered.log
    Untracked:  output/chr4_DCas21_5841_WA_REFimputedAndFiltered.nosex
    Untracked:  output/chr4_DCas21_5841_WA_REFimputedAndFiltered.raw
    Untracked:  output/chr4_DCas21_5841_WA_REFimputedAndFiltered.sitesWithAlleles
    Untracked:  output/chr4_DCas21_5841_WA_REFimputedAndFiltered.vcf.gz
    Untracked:  output/chr5_DCas21_5841_WA_REFimputed.INFO
    Untracked:  output/chr5_DCas21_5841_WA_REFimputed.hwe
    Untracked:  output/chr5_DCas21_5841_WA_REFimputed.log
    Untracked:  output/chr5_DCas21_5841_WA_REFimputed.sitesPassing
    Untracked:  output/chr5_DCas21_5841_WA_REFimputed.vcf.gz
    Untracked:  output/chr5_DCas21_5841_WA_REFimputedAndFiltered.alleleToCount
    Untracked:  output/chr5_DCas21_5841_WA_REFimputedAndFiltered.bed
    Untracked:  output/chr5_DCas21_5841_WA_REFimputedAndFiltered.bim
    Untracked:  output/chr5_DCas21_5841_WA_REFimputedAndFiltered.fam
    Untracked:  output/chr5_DCas21_5841_WA_REFimputedAndFiltered.log
    Untracked:  output/chr5_DCas21_5841_WA_REFimputedAndFiltered.nosex
    Untracked:  output/chr5_DCas21_5841_WA_REFimputedAndFiltered.raw
    Untracked:  output/chr5_DCas21_5841_WA_REFimputedAndFiltered.sitesWithAlleles
    Untracked:  output/chr5_DCas21_5841_WA_REFimputedAndFiltered.vcf.gz
    Untracked:  output/chr6_DCas21_5841_WA_REFimputed.INFO
    Untracked:  output/chr6_DCas21_5841_WA_REFimputed.hwe
    Untracked:  output/chr6_DCas21_5841_WA_REFimputed.log
    Untracked:  output/chr6_DCas21_5841_WA_REFimputed.sitesPassing
    Untracked:  output/chr6_DCas21_5841_WA_REFimputed.vcf.gz
    Untracked:  output/chr6_DCas21_5841_WA_REFimputedAndFiltered.alleleToCount
    Untracked:  output/chr6_DCas21_5841_WA_REFimputedAndFiltered.bed
    Untracked:  output/chr6_DCas21_5841_WA_REFimputedAndFiltered.bim
    Untracked:  output/chr6_DCas21_5841_WA_REFimputedAndFiltered.fam
    Untracked:  output/chr6_DCas21_5841_WA_REFimputedAndFiltered.log
    Untracked:  output/chr6_DCas21_5841_WA_REFimputedAndFiltered.nosex
    Untracked:  output/chr6_DCas21_5841_WA_REFimputedAndFiltered.raw
    Untracked:  output/chr6_DCas21_5841_WA_REFimputedAndFiltered.sitesWithAlleles
    Untracked:  output/chr6_DCas21_5841_WA_REFimputedAndFiltered.vcf.gz
    Untracked:  output/chr7_DCas21_5841_WA_REFimputed.INFO
    Untracked:  output/chr7_DCas21_5841_WA_REFimputed.hwe
    Untracked:  output/chr7_DCas21_5841_WA_REFimputed.log
    Untracked:  output/chr7_DCas21_5841_WA_REFimputed.sitesPassing
    Untracked:  output/chr7_DCas21_5841_WA_REFimputed.vcf.gz
    Untracked:  output/chr7_DCas21_5841_WA_REFimputedAndFiltered.alleleToCount
    Untracked:  output/chr7_DCas21_5841_WA_REFimputedAndFiltered.bed
    Untracked:  output/chr7_DCas21_5841_WA_REFimputedAndFiltered.bim
    Untracked:  output/chr7_DCas21_5841_WA_REFimputedAndFiltered.fam
    Untracked:  output/chr7_DCas21_5841_WA_REFimputedAndFiltered.log
    Untracked:  output/chr7_DCas21_5841_WA_REFimputedAndFiltered.nosex
    Untracked:  output/chr7_DCas21_5841_WA_REFimputedAndFiltered.raw
    Untracked:  output/chr7_DCas21_5841_WA_REFimputedAndFiltered.sitesWithAlleles
    Untracked:  output/chr7_DCas21_5841_WA_REFimputedAndFiltered.vcf.gz
    Untracked:  output/chr8_DCas21_5841_WA_REFimputed.INFO
    Untracked:  output/chr8_DCas21_5841_WA_REFimputed.hwe
    Untracked:  output/chr8_DCas21_5841_WA_REFimputed.log
    Untracked:  output/chr8_DCas21_5841_WA_REFimputed.sitesPassing
    Untracked:  output/chr8_DCas21_5841_WA_REFimputed.vcf.gz
    Untracked:  output/chr8_DCas21_5841_WA_REFimputedAndFiltered.alleleToCount
    Untracked:  output/chr8_DCas21_5841_WA_REFimputedAndFiltered.bed
    Untracked:  output/chr8_DCas21_5841_WA_REFimputedAndFiltered.bim
    Untracked:  output/chr8_DCas21_5841_WA_REFimputedAndFiltered.fam
    Untracked:  output/chr8_DCas21_5841_WA_REFimputedAndFiltered.log
    Untracked:  output/chr8_DCas21_5841_WA_REFimputedAndFiltered.nosex
    Untracked:  output/chr8_DCas21_5841_WA_REFimputedAndFiltered.raw
    Untracked:  output/chr8_DCas21_5841_WA_REFimputedAndFiltered.sitesWithAlleles
    Untracked:  output/chr8_DCas21_5841_WA_REFimputedAndFiltered.vcf.gz
    Untracked:  output/chr9_DCas21_5841_WA_REFimputed.INFO
    Untracked:  output/chr9_DCas21_5841_WA_REFimputed.hwe
    Untracked:  output/chr9_DCas21_5841_WA_REFimputed.log
    Untracked:  output/chr9_DCas21_5841_WA_REFimputed.sitesPassing
    Untracked:  output/chr9_DCas21_5841_WA_REFimputed.vcf.gz
    Untracked:  output/chr9_DCas21_5841_WA_REFimputedAndFiltered.alleleToCount
    Untracked:  output/chr9_DCas21_5841_WA_REFimputedAndFiltered.bed
    Untracked:  output/chr9_DCas21_5841_WA_REFimputedAndFiltered.bim
    Untracked:  output/chr9_DCas21_5841_WA_REFimputedAndFiltered.fam
    Untracked:  output/chr9_DCas21_5841_WA_REFimputedAndFiltered.log
    Untracked:  output/chr9_DCas21_5841_WA_REFimputedAndFiltered.nosex
    Untracked:  output/chr9_DCas21_5841_WA_REFimputedAndFiltered.raw
    Untracked:  output/chr9_DCas21_5841_WA_REFimputedAndFiltered.sitesWithAlleles
    Untracked:  output/chr9_DCas21_5841_WA_REFimputedAndFiltered.vcf.gz
    Untracked:  output/cvresults_ADE_1_2021May03.rds
    Untracked:  output/cvresults_ADE_2_2021May03.rds
    Untracked:  output/cvresults_A_2021May03.rds
    Untracked:  output/genomicPredictions_ModelADE_twostage_NRCRI_2021May03.rds
    Untracked:  output/genomicPredictions_ModelA_twostage_NRCRI_2021May03.rds
    Untracked:  output/genomicPredictions_NRCRI_2021May03.csv
    Untracked:  output/maxNOHAV_byStudy.csv

Unstaged changes:
    Modified:   README.md

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/03-CrossValidation.Rmd</code>) and HTML (<code>docs/03-CrossValidation.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/wolfemd/NRCRI_2021GS/blob/94ec81122c0b7a209fbf75bf47ca3e2101a0f529/analysis/03-CrossValidation.Rmd" target="_blank">94ec811</a>
</td>
<td>
wolfemd
</td>
<td>
2021-05-05
</td>
<td>
Completed prediction. Build site and push to GitHub / Cassavabase FTP server. Share to NRCRI.
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="previous-step" class="section level1">
<h1>Previous step</h1>
<ol start="2" style="list-style-type: decimal">
<li><a href="02-GetBLUPs.html">Get BLUPs combining all trial data</a>: Combine data from all trait-trials to get BLUPs for downstream genomic prediction.
<ul>
<li>Fit mixed-model to multi-trial dataset and extract BLUPs, de-regressed BLUPs and weights. Include two rounds of outlier removal.</li>
</ul></li>
</ol>
</div>
<div id="cross-validation" class="section level1">
<h1>Cross-validation</h1>
<p>5-fold cross-validation. Replicate 5-times.</p>
<p>2 genomic models:</p>
<ol style="list-style-type: decimal">
<li>Additive-only (<strong>A</strong>)</li>
<li>Addititive plus dominance plus additive-by-dominance epistasis (<strong>ADE</strong>)</li>
</ol>
</div>
<div id="prep.-genomic-data" class="section level1">
<h1>Prep. genomic data</h1>
<div id="get-snp-data-from-ftp" class="section level2">
<h2>Get SNP data from FTP</h2>
<p>The data for the next step can be found on the cassavabase FTP server <a href="ftp://ftp.cassavabase.org/marnin_datasets/NRCRI_2021GS/">here</a>.</p>
<p>Can be loaded directly to R from FTP.</p>
<p><strong>NOTICE:</strong> You need enough RAM and a stable network connection. I do the next steps, including cross-validation on a server with plenty of RAM and a good, stable network connection, rather than on my personal computer (a laptop with 16 GB RAM).</p>
<p>The outputs (kinship matrices and filtered snp dosages) of the steps below, which are too large for GitHub, can be found on the cassavabase FTP server <a href="ftp://ftp.cassavabase.org/marnin_datasets/NRCRI_2021GS/">here</a>.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="co"># activate multithread OpenBLAS for fast compute of SigmaM (genotypic var-covar matrix)</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> /home/jj332_cas/marnin/NRCRI_2021GS/<span class="kw">;</span> </span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="bu">export</span> <span class="va">OMP_NUM_THREADS=</span>56</span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse); <span class="fu">library</span>(magrittr); </span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a><span class="co"># RefPanel (including primary TP and original GS progeny)</span></span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Dosage matrix</span></span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a>snps_refpanel<span class="ot">&lt;-</span><span class="fu">readRDS</span>(<span class="at">file=</span><span class="fu">url</span>(<span class="fu">paste0</span>(<span class="st">&quot;ftp://ftp.cassavabase.org/marnin_datasets/NGC_BigData/&quot;</span>,</span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a>                              <span class="st">&quot;DosageMatrix_RefPanelAndGSprogeny_ReadyForGP_73019.rds&quot;</span>)))</span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a><span class="co"># 2020 GS progeny dosages (C3a)</span></span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a><span class="do">## these don&#39;t have phenotypes yet. including for convenience </span></span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a><span class="do">## and so we don&#39;t get suprised by their absence in the next iteration</span></span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a>snps5510<span class="ot">&lt;-</span><span class="fu">readRDS</span>(<span class="at">file=</span><span class="fu">url</span>(<span class="fu">paste0</span>(<span class="st">&quot;ftp://ftp.cassavabase.org/marnin_datasets/NRCRI_2020GS/output/&quot;</span>,</span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a>                                  <span class="st">&quot;DosageMatrix_DCas20_5510_WA_REFimputedAndFiltered.rds&quot;</span>)))</span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a>snps5440<span class="ot">&lt;-</span><span class="fu">readRDS</span>(<span class="at">file=</span><span class="fu">url</span>(<span class="fu">paste0</span>(<span class="st">&quot;ftp://ftp.cassavabase.org/marnin_datasets/NRCRI_2020GS/output/&quot;</span>,</span>
<span id="cb2-12"><a href="#cb2-12" aria-hidden="true" tabindex="-1"></a>                                  <span class="st">&quot;DosageMatrix_DCas20_5440_WA_REFimputedAndFiltered.rds&quot;</span>)))</span>
<span id="cb2-13"><a href="#cb2-13" aria-hidden="true" tabindex="-1"></a><span class="do">## DCas21-5841 Dosage matrix (C3b)</span></span>
<span id="cb2-14"><a href="#cb2-14" aria-hidden="true" tabindex="-1"></a>snps5841<span class="ot">&lt;-</span><span class="fu">readRDS</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;output&quot;</span>,<span class="st">&quot;DosageMatrix_DCas21_5841_WA_REFimputedAndFiltered.rds&quot;</span>))</span>
<span id="cb2-15"><a href="#cb2-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-16"><a href="#cb2-16" aria-hidden="true" tabindex="-1"></a>snps2keep<span class="ot">&lt;-</span><span class="fu">colnames</span>(snps_refpanel) <span class="sc">%&gt;%</span> </span>
<span id="cb2-17"><a href="#cb2-17" aria-hidden="true" tabindex="-1"></a>  .[. <span class="sc">%in%</span> <span class="fu">colnames</span>(snps5841)] <span class="sc">%&gt;%</span> </span>
<span id="cb2-18"><a href="#cb2-18" aria-hidden="true" tabindex="-1"></a>  .[. <span class="sc">%in%</span> <span class="fu">colnames</span>(snps5510)] <span class="sc">%&gt;%</span> </span>
<span id="cb2-19"><a href="#cb2-19" aria-hidden="true" tabindex="-1"></a>  .[. <span class="sc">%in%</span> <span class="fu">colnames</span>(snps5440)]</span>
<span id="cb2-20"><a href="#cb2-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-21"><a href="#cb2-21" aria-hidden="true" tabindex="-1"></a>snps<span class="ot">&lt;-</span><span class="fu">rbind</span>(snps_refpanel[,snps2keep],</span>
<span id="cb2-22"><a href="#cb2-22" aria-hidden="true" tabindex="-1"></a>            snps5510[,snps2keep]) <span class="sc">%&gt;%</span> </span>
<span id="cb2-23"><a href="#cb2-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rbind</span>(.,snps5440[,snps2keep]) <span class="sc">%&gt;%</span> </span>
<span id="cb2-24"><a href="#cb2-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rbind</span>(.,snps5841[,snps2keep])</span>
<span id="cb2-25"><a href="#cb2-25" aria-hidden="true" tabindex="-1"></a><span class="fu">gc</span>()</span>
<span id="cb2-26"><a href="#cb2-26" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(snps) <span class="co"># [1] 22936 20559</span></span>
<span id="cb2-27"><a href="#cb2-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-28"><a href="#cb2-28" aria-hidden="true" tabindex="-1"></a><span class="co">#rm(list=(ls() %&gt;% grep(&quot;snps&quot;,.,value = T, invert = T)))</span></span></code></pre></div>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>blups<span class="ot">&lt;-</span><span class="fu">readRDS</span>(<span class="at">file=</span>here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;output&quot;</span>,<span class="st">&quot;NRCRI_blupsForModelTraining_twostage_asreml_2021May03.rds&quot;</span>))</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>blups <span class="sc">%&lt;&gt;%</span> </span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(Trait,blups) <span class="sc">%&gt;%</span> </span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unnest</span>(blups) <span class="sc">%&gt;%</span> </span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span><span class="st">`</span><span class="at">std error</span><span class="st">`</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(GID <span class="sc">%in%</span> <span class="fu">rownames</span>(snps))</span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(<span class="fu">unique</span>(blups<span class="sc">$</span>GID) <span class="sc">%in%</span> <span class="fu">rownames</span>(snps)) </span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a><span class="co"># TRUE</span></span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a><span class="co"># 3114</span></span></code></pre></div>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>samples2Keep<span class="ot">&lt;-</span><span class="fu">unique</span>(blups<span class="sc">$</span>GID) <span class="sc">%&gt;%</span> </span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">union</span>(.,<span class="fu">grep</span>(<span class="st">&quot;c2|c3&quot;</span>,<span class="fu">rownames</span>(snps),<span class="at">value =</span> T, <span class="at">ignore.case =</span> T)) <span class="sc">%&gt;%</span> </span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">union</span>(.,<span class="fu">rownames</span>(snps5510)) <span class="sc">%&gt;%</span> </span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">union</span>(.,<span class="fu">rownames</span>(snps5440)) <span class="sc">%&gt;%</span> </span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">union</span>(.,<span class="fu">rownames</span>(snps5841))</span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(samples2Keep) <span class="co"># [1] 7286 </span></span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a>snps<span class="ot">&lt;-</span>snps[samples2Keep,]; </span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a><span class="fu">gc</span>()</span></code></pre></div>
</div>
<div id="maf1-filter" class="section level2">
<h2>MAF&gt;1% filter</h2>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;code&quot;</span>,<span class="st">&quot;gsFunctions.R&quot;</span>))</span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>snps <span class="sc">%&lt;&gt;%</span> <span class="fu">maf_filter</span>(.,<span class="fl">0.01</span>)</span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(snps) <span class="co"># [1]  7286 20550</span></span></code></pre></div>
</div>
<div id="make-add-dom-and-epi-kinships" class="section level2">
<h2>Make Add, Dom and Epi kinships</h2>
<p>Going to use my own kinship function.</p>
<p>Make the kinships.</p>
<p>Below e.g. <code>A*A</code> makes a matrix that approximates additive-by-additive epistasis relationships.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>A<span class="ot">&lt;-</span><span class="fu">kinship</span>(snps,<span class="at">type=</span><span class="st">&quot;add&quot;</span>)</span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>D<span class="ot">&lt;-</span><span class="fu">kinship</span>(snps,<span class="at">type=</span><span class="st">&quot;dom&quot;</span>)</span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>AD<span class="ot">&lt;-</span>A<span class="sc">*</span>D</span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(snps,<span class="at">file=</span>here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;output&quot;</span>,<span class="st">&quot;DosageMatrix_NRCRI_2021May03.rds&quot;</span>))</span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(A,<span class="at">file=</span>here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;output&quot;</span>,<span class="st">&quot;Kinship_A_NRCRI_2021May03.rds&quot;</span>))</span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(D,<span class="at">file=</span>here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;output&quot;</span>,<span class="st">&quot;Kinship_D_NRCRI_2021May03.rds&quot;</span>))</span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(AD,<span class="at">file=</span>here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;output&quot;</span>,<span class="st">&quot;Kinship_AD_NRCRI_2021May03.rds&quot;</span>))</span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a><span class="co">#rm(snps); gc()</span></span></code></pre></div>
<p><strong>NOTICE:</strong> The outputs (kinship matrices and filtered snp dosages) of the steps below, which are too large for GitHub, can be found on the cassavabase FTP server <a href="ftp://ftp.cassavabase.org/marnin_datasets/NRCRI_2021GS/">here</a>.</p>
</div>
</div>
<div id="cross-validation-1" class="section level1">
<h1>Cross-validation</h1>
<div class="sourceCode" id="cb7"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> /home/jj332_cas/marnin/NRCRI_2021GS/<span class="kw">;</span> </span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a><span class="bu">export</span> <span class="va">OMP_NUM_THREADS=</span>56 # <span class="va">activate</span> <span class="va">multithread</span> <span class="va">OpenBLAS</span> </span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a><span class="co">##### [considered] </span></span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a><span class="co">######/programs/R-4.0.0/bin/R # switched to R V4, having trouble with sommer/Matrix in v3.5</span></span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a><span class="co">## may have to reinstall packages</span></span></code></pre></div>
<div id="set-up-training-testing-data" class="section level2">
<h2>Set-up training-testing data</h2>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list=</span><span class="fu">ls</span>())</span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse); <span class="fu">library</span>(magrittr); </span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;code&quot;</span>,<span class="st">&quot;gsFunctions.R&quot;</span>))</span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>blups<span class="ot">&lt;-</span><span class="fu">readRDS</span>(<span class="at">file=</span>here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;output&quot;</span>,<span class="st">&quot;NRCRI_blupsForModelTraining_twostage_asreml_2021May03.rds&quot;</span>))</span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a>A<span class="ot">&lt;-</span><span class="fu">readRDS</span>(<span class="at">file=</span>here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;output&quot;</span>,<span class="st">&quot;Kinship_A_NRCRI_2021May03.rds&quot;</span>))</span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a>blups <span class="sc">%&lt;&gt;%</span> </span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(Trait,blups) <span class="sc">%&gt;%</span> </span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unnest</span>(blups) <span class="sc">%&gt;%</span> </span>
<span id="cb8-10"><a href="#cb8-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span><span class="st">`</span><span class="at">std error</span><span class="st">`</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb8-11"><a href="#cb8-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(GID <span class="sc">%in%</span> <span class="fu">rownames</span>(A))</span>
<span id="cb8-12"><a href="#cb8-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-13"><a href="#cb8-13" aria-hidden="true" tabindex="-1"></a>cv2do<span class="ot">&lt;-</span>blups <span class="sc">%&gt;%</span></span>
<span id="cb8-14"><a href="#cb8-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">nest</span>(<span class="at">TrainTestData=</span><span class="sc">-</span>Trait)</span>
<span id="cb8-15"><a href="#cb8-15" aria-hidden="true" tabindex="-1"></a>cv2do <span class="sc">%&gt;%</span> rmarkdown<span class="sc">::</span><span class="fu">paged_table</span>()</span></code></pre></div>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["Trait"],"name":[1],"type":["chr"],"align":["left"]},{"label":["TrainTestData"],"name":[2],"type":["list"],"align":["right"]}],"data":[{"1":"CGMS1","2":"<tibble[,6]>"},{"1":"CGMS2","2":"<tibble[,6]>"},{"1":"MCMDS","2":"<tibble[,6]>"},{"1":"DMsg","2":"<tibble[,6]>"},{"1":"PLTHT","2":"<tibble[,6]>"},{"1":"BRNHT1","2":"<tibble[,6]>"},{"1":"HI","2":"<tibble[,6]>"},{"1":"logFYLD","2":"<tibble[,6]>"},{"1":"logTOPYLD","2":"<tibble[,6]>"},{"1":"logRTNO","2":"<tibble[,6]>"},{"1":"DM","2":"<tibble[,6]>"},{"1":"logDYLD","2":"<tibble[,6]>"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="co"># # A tibble: 12 x 2</span></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a><span class="co">#    Trait     TrainTestData     </span></span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a><span class="co">#    &lt;chr&gt;     &lt;list&gt;            </span></span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a><span class="co">#  1 CGMS1     &lt;tibble [2,986 × 6]&gt;</span></span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a><span class="co">#  2 CGMS2     &lt;tibble [2,144 × 6]&gt;</span></span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a><span class="co">#  3 MCMDS     &lt;tibble [2,987 × 6]&gt;</span></span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a><span class="co">#  4 DMsg      &lt;tibble [791 × 6]&gt;  </span></span>
<span id="cb9-8"><a href="#cb9-8" aria-hidden="true" tabindex="-1"></a><span class="co">#  5 PLTHT     &lt;tibble [2,505 × 6]&gt;</span></span>
<span id="cb9-9"><a href="#cb9-9" aria-hidden="true" tabindex="-1"></a><span class="co">#  6 BRNHT1    &lt;tibble [1,860 × 6]&gt;</span></span>
<span id="cb9-10"><a href="#cb9-10" aria-hidden="true" tabindex="-1"></a><span class="co">#  7 HI        &lt;tibble [2,711 × 6]&gt;</span></span>
<span id="cb9-11"><a href="#cb9-11" aria-hidden="true" tabindex="-1"></a><span class="co">#  8 logFYLD   &lt;tibble [2,688 × 6]&gt;</span></span>
<span id="cb9-12"><a href="#cb9-12" aria-hidden="true" tabindex="-1"></a><span class="co">#  9 logTOPYLD &lt;tibble [2,702 × 6]&gt;</span></span>
<span id="cb9-13"><a href="#cb9-13" aria-hidden="true" tabindex="-1"></a><span class="co"># 10 logRTNO   &lt;tibble [2,693 × 6]&gt;</span></span>
<span id="cb9-14"><a href="#cb9-14" aria-hidden="true" tabindex="-1"></a><span class="co"># 11 DM        &lt;tibble [2,130 × 6]&gt;</span></span>
<span id="cb9-15"><a href="#cb9-15" aria-hidden="true" tabindex="-1"></a><span class="co"># 12 logDYLD   &lt;tibble [2,064 × 6]&gt;</span></span></code></pre></div>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>cv2do<span class="sc">$</span>TrainTestData[[<span class="dv">6</span>]] <span class="sc">%&gt;%</span> head <span class="sc">%&gt;%</span> rmarkdown<span class="sc">::</span><span class="fu">paged_table</span>()</span></code></pre></div>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["GID"],"name":[1],"type":["chr"],"align":["left"]},{"label":["BLUP"],"name":[2],"type":["dbl"],"align":["right"]},{"label":["PEV"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["REL"],"name":[4],"type":["dbl"],"align":["right"]},{"label":["drgBLUP"],"name":[5],"type":["dbl"],"align":["right"]},{"label":["WT"],"name":[6],"type":["dbl"],"align":["right"]}],"data":[{"1":"AR124:250107818","2":"8.647876","3":"46.46010","4":"0.4054099","5":"21.331190","6":"7.276556"},{"1":"AR1410:250399710","2":"4.063514","3":"44.17898","4":"0.4346033","5":"9.349938","6":"8.137156"},{"1":"AR144:250107805","2":"-5.134758","3":"44.26454","4":"0.4335084","5":"-11.844655","6":"8.103540"},{"1":"AR155:250134515","2":"-6.878125","3":"54.83950","4":"0.2981715","5":"-23.067685","6":"4.645796"},{"1":"AR182:250134520","2":"-2.311664","3":"62.32273","4":"0.2024021","5":"-11.421147","6":"2.821257"},{"1":"AR311:250134529","2":"4.939141","3":"58.27090","4":"0.2542569","5":"19.425787","6":"3.758532"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="co">#                         GID       BLUP       PEV       REL    drgBLUP       WT</span></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a><span class="co"># 1  AR124:250107818  8.647876 46.46010 0.4054099  21.331190 7.276556</span></span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a><span class="co"># 2 AR1410:250399710  4.063514 44.17898 0.4346033   9.349938 8.137156</span></span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a><span class="co"># 3  AR144:250107805 -5.134758 44.26454 0.4335084 -11.844655 8.103540</span></span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a><span class="co"># 4  AR155:250134515 -6.878125 54.83950 0.2981715 -23.067685 4.645796</span></span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a><span class="co"># 5  AR182:250134520 -2.311664 62.32273 0.2024021 -11.421147 2.821257</span></span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a><span class="co"># 6  AR311:250134529  4.939141 58.27090 0.2542569  19.425787 3.758532</span></span></code></pre></div>
<p>The function below <code>runCrossVal()</code> function implements nfold cross-validation. Specifically, for each of <strong>nrepeats</strong> it splits the data into <strong>nfolds</strong> sets according to <strong>gid</strong>. So if <code>nfolds=5</code> then the the clones will be divided into 5 groups and 5 predictions will be made. In each prediction, 4/5 of the clones will be used to predict the remaining 1/5. Accuracy of the model is measured as the correlation between the BLUPs (adj. mean for each CLONE) in the <em>test set</em> and the GEBV (the prediction made of each clone when it was in the test set).</p>
<p>Below, 20 reps x 5-fold cross-validation are run on 1 large memory Cornell CBSU machine each (e.g. <a href="https://biohpc.cornell.edu/lab/hardware.aspx">cbsulm29; 104 cores, 512 GB RAM</a>).</p>
</div>
<div id="cv---modeltypea" class="section level2">
<h2>CV - modelType=“A”</h2>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a>starttime<span class="ot">&lt;-</span><span class="fu">proc.time</span>()[<span class="dv">3</span>]</span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>cv_A<span class="ot">&lt;-</span>cv2do <span class="sc">%&gt;%</span> </span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">CVresults=</span><span class="fu">map</span>(TrainTestData,<span class="sc">~</span><span class="fu">runCrossVal</span>(<span class="at">TrainTestData=</span>.,</span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a>                                                  <span class="at">modelType=</span><span class="st">&quot;A&quot;</span>,</span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a>                                                  <span class="at">grms=</span><span class="fu">list</span>(<span class="at">A=</span>A),</span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a>                                                  <span class="at">byGroup=</span><span class="cn">FALSE</span>,<span class="at">augmentTP=</span><span class="cn">NULL</span>,</span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a>                                                  <span class="at">nrepeats=</span><span class="dv">20</span>,<span class="at">nfolds=</span><span class="dv">5</span>,<span class="at">ncores=</span><span class="dv">25</span>,<span class="at">gid=</span><span class="st">&quot;GID&quot;</span>)))</span>
<span id="cb12-8"><a href="#cb12-8" aria-hidden="true" tabindex="-1"></a>runtime<span class="ot">&lt;-</span><span class="fu">proc.time</span>()[<span class="dv">3</span>]<span class="sc">-</span>starttime; runtime</span>
<span id="cb12-9"><a href="#cb12-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-10"><a href="#cb12-10" aria-hidden="true" tabindex="-1"></a>cv_A <span class="sc">%&lt;&gt;%</span> <span class="fu">mutate</span>(<span class="at">modelType=</span><span class="st">&quot;A&quot;</span>) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>TrainTestData)</span>
<span id="cb12-11"><a href="#cb12-11" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(cv_A,<span class="at">file=</span>here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;output&quot;</span>,<span class="st">&quot;cvresults_A_2021May03.rds&quot;</span>))</span></code></pre></div>
</div>
<div id="cv---modeltypeade" class="section level2">
<h2>CV - modelType=“ADE”</h2>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="co"># split it to two tranches of 10 reps</span></span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a><span class="fu">options</span>(<span class="at">future.globals.maxSize=</span> <span class="dv">3000</span><span class="sc">*</span><span class="dv">1024</span><span class="sc">^</span><span class="dv">2</span>)</span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a>D<span class="ot">&lt;-</span><span class="fu">readRDS</span>(<span class="at">file=</span>here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;output&quot;</span>,<span class="st">&quot;Kinship_D_NRCRI_2021May03.rds&quot;</span>))</span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a>AD<span class="ot">&lt;-</span><span class="fu">readRDS</span>(<span class="at">file=</span>here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;output&quot;</span>,<span class="st">&quot;Kinship_AD_NRCRI_2021May03.rds&quot;</span>))</span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a>starttime<span class="ot">&lt;-</span><span class="fu">proc.time</span>()[<span class="dv">3</span>]</span>
<span id="cb13-7"><a href="#cb13-7" aria-hidden="true" tabindex="-1"></a>cv_ADE_1<span class="ot">&lt;-</span>cv2do <span class="sc">%&gt;%</span> </span>
<span id="cb13-8"><a href="#cb13-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">CVresults=</span><span class="fu">map</span>(TrainTestData,<span class="sc">~</span><span class="fu">runCrossVal</span>(<span class="at">TrainTestData=</span>.,</span>
<span id="cb13-9"><a href="#cb13-9" aria-hidden="true" tabindex="-1"></a>                                                  <span class="at">modelType=</span><span class="st">&quot;ADE&quot;</span>,</span>
<span id="cb13-10"><a href="#cb13-10" aria-hidden="true" tabindex="-1"></a>                                                  <span class="at">grms=</span><span class="fu">list</span>(<span class="at">A=</span>A,<span class="at">D=</span>D,<span class="at">AD=</span>AD),</span>
<span id="cb13-11"><a href="#cb13-11" aria-hidden="true" tabindex="-1"></a>                                                  <span class="at">byGroup=</span><span class="cn">FALSE</span>,<span class="at">augmentTP=</span><span class="cn">NULL</span>,</span>
<span id="cb13-12"><a href="#cb13-12" aria-hidden="true" tabindex="-1"></a>                                                  <span class="at">nrepeats=</span><span class="dv">10</span>,<span class="at">nfolds=</span><span class="dv">5</span>,<span class="at">ncores=</span><span class="dv">5</span>,<span class="at">gid=</span><span class="st">&quot;GID&quot;</span>)))</span>
<span id="cb13-13"><a href="#cb13-13" aria-hidden="true" tabindex="-1"></a>cv_ADE_1 <span class="sc">%&lt;&gt;%</span> <span class="fu">mutate</span>(<span class="at">modelType=</span><span class="st">&quot;ADE&quot;</span>) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>TrainTestData)</span>
<span id="cb13-14"><a href="#cb13-14" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(cv_ADE_1,<span class="at">file=</span>here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;output&quot;</span>,<span class="st">&quot;cvresults_ADE_1_2021May03.rds&quot;</span>))</span>
<span id="cb13-15"><a href="#cb13-15" aria-hidden="true" tabindex="-1"></a>runtime<span class="ot">&lt;-</span><span class="fu">proc.time</span>()[<span class="dv">3</span>]<span class="sc">-</span>starttime; runtime</span>
<span id="cb13-16"><a href="#cb13-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-17"><a href="#cb13-17" aria-hidden="true" tabindex="-1"></a>starttime<span class="ot">&lt;-</span><span class="fu">proc.time</span>()[<span class="dv">3</span>]</span>
<span id="cb13-18"><a href="#cb13-18" aria-hidden="true" tabindex="-1"></a>cv_ADE_2<span class="ot">&lt;-</span>cv2do <span class="sc">%&gt;%</span> </span>
<span id="cb13-19"><a href="#cb13-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">CVresults=</span><span class="fu">map</span>(TrainTestData,<span class="sc">~</span><span class="fu">runCrossVal</span>(<span class="at">TrainTestData=</span>.,</span>
<span id="cb13-20"><a href="#cb13-20" aria-hidden="true" tabindex="-1"></a>                                                  <span class="at">modelType=</span><span class="st">&quot;ADE&quot;</span>,</span>
<span id="cb13-21"><a href="#cb13-21" aria-hidden="true" tabindex="-1"></a>                                                  <span class="at">grms=</span><span class="fu">list</span>(<span class="at">A=</span>A,<span class="at">D=</span>D,<span class="at">AD=</span>AD),</span>
<span id="cb13-22"><a href="#cb13-22" aria-hidden="true" tabindex="-1"></a>                                                  <span class="at">byGroup=</span><span class="cn">FALSE</span>,<span class="at">augmentTP=</span><span class="cn">NULL</span>,</span>
<span id="cb13-23"><a href="#cb13-23" aria-hidden="true" tabindex="-1"></a>                                                  <span class="at">nrepeats=</span><span class="dv">10</span>,<span class="at">nfolds=</span><span class="dv">5</span>,<span class="at">ncores=</span><span class="dv">5</span>,<span class="at">gid=</span><span class="st">&quot;GID&quot;</span>)))</span>
<span id="cb13-24"><a href="#cb13-24" aria-hidden="true" tabindex="-1"></a>cv_ADE_2 <span class="sc">%&lt;&gt;%</span> <span class="fu">mutate</span>(<span class="at">modelType=</span><span class="st">&quot;ADE&quot;</span>) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>TrainTestData)</span>
<span id="cb13-25"><a href="#cb13-25" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(cv_ADE_2,<span class="at">file=</span>here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;output&quot;</span>,<span class="st">&quot;cvresults_ADE_2_2021May03.rds&quot;</span>))</span>
<span id="cb13-26"><a href="#cb13-26" aria-hidden="true" tabindex="-1"></a>runtime<span class="ot">&lt;-</span><span class="fu">proc.time</span>()[<span class="dv">3</span>]<span class="sc">-</span>starttime; runtime</span></code></pre></div>
</div>
</div>
<div id="results" class="section level1">
<h1>Results</h1>
<p>See <a href="05-Results.html">Results</a>: Home for plots and summary tables.</p>
</div>
<div id="next-step" class="section level1">
<h1>Next step</h1>
<ol start="4" style="list-style-type: decimal">
<li><a href="04-GetGBLUPs.html">Genomic prediction</a>: Predict <em>genomic</em> BLUPs (GEBV and GETGV) for all selection candidates using all available data.</li>
</ol>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code></pre></div>
<pre><code>R version 4.0.3 (2020-10-10)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] magrittr_2.0.1  forcats_0.5.1   stringr_1.4.0   dplyr_1.0.5    
 [5] purrr_0.3.4     readr_1.4.0     tidyr_1.1.3     tibble_3.1.1   
 [9] ggplot2_3.3.3   tidyverse_1.3.1 workflowr_1.6.2

loaded via a namespace (and not attached):
 [1] tidyselect_1.1.0  xfun_0.22         bslib_0.2.4       haven_2.4.0      
 [5] colorspace_2.0-0  vctrs_0.3.7       generics_0.1.0    htmltools_0.5.1.1
 [9] yaml_2.2.1        utf8_1.2.1        rlang_0.4.10      jquerylib_0.1.3  
[13] later_1.1.0.1     pillar_1.6.0      withr_2.4.2       glue_1.4.2       
[17] DBI_1.1.1         dbplyr_2.1.1      readxl_1.3.1      modelr_0.1.8     
[21] lifecycle_1.0.0   cellranger_1.1.0  munsell_0.5.0     gtable_0.3.0     
[25] rvest_1.0.0       evaluate_0.14     knitr_1.32        httpuv_1.5.5     
[29] fansi_0.4.2       broom_0.7.6       Rcpp_1.0.6        promises_1.2.0.1 
[33] backports_1.2.1   scales_1.1.1      jsonlite_1.7.2    fs_1.5.0         
[37] hms_1.0.0         digest_0.6.27     stringi_1.5.3     rprojroot_2.0.2  
[41] grid_4.0.3        here_1.0.1        cli_2.4.0         tools_4.0.3      
[45] sass_0.3.1        crayon_1.4.1      whisker_0.4       pkgconfig_2.0.3  
[49] ellipsis_0.3.1    xml2_1.3.2        reprex_2.0.0      lubridate_1.7.10 
[53] rstudioapi_0.13   assertthat_0.2.1  rmarkdown_2.7     httr_1.4.2       
[57] R6_2.5.0          git2r_0.28.0      compiler_4.0.3   </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
