<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>NRCRI 2021 GS and related procedures</title>

<script src="site_libs/header-attrs-2.7/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/yeti.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>


<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    background-color: #232629;
    color: #7a7c7d;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #7a7c7d;  padding-left: 4px; }
div.sourceCode
  { color: #cfcfc2; background-color: #232629; }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span { color: #cfcfc2; } /* Normal */
code span.al { color: #95da4c; background-color: #4d1f24; font-weight: bold; } /* Alert */
code span.an { color: #3f8058; } /* Annotation */
code span.at { color: #2980b9; } /* Attribute */
code span.bn { color: #f67400; } /* BaseN */
code span.bu { color: #7f8c8d; } /* BuiltIn */
code span.cf { color: #fdbc4b; font-weight: bold; } /* ControlFlow */
code span.ch { color: #3daee9; } /* Char */
code span.cn { color: #27aeae; font-weight: bold; } /* Constant */
code span.co { color: #7a7c7d; } /* Comment */
code span.cv { color: #7f8c8d; } /* CommentVar */
code span.do { color: #a43340; } /* Documentation */
code span.dt { color: #2980b9; } /* DataType */
code span.dv { color: #f67400; } /* DecVal */
code span.er { color: #da4453; text-decoration: underline; } /* Error */
code span.ex { color: #0099ff; font-weight: bold; } /* Extension */
code span.fl { color: #f67400; } /* Float */
code span.fu { color: #8e44ad; } /* Function */
code span.im { color: #27ae60; } /* Import */
code span.in { color: #c45b00; } /* Information */
code span.kw { color: #cfcfc2; font-weight: bold; } /* Keyword */
code span.op { color: #cfcfc2; } /* Operator */
code span.ot { color: #27ae60; } /* Other */
code span.pp { color: #27ae60; } /* Preprocessor */
code span.re { color: #2980b9; background-color: #153042; } /* RegionMarker */
code span.sc { color: #3daee9; } /* SpecialChar */
code span.ss { color: #da4453; } /* SpecialString */
code span.st { color: #f44f4f; } /* String */
code span.va { color: #27aeae; } /* Variable */
code span.vs { color: #da4453; } /* VerbatimString */
code span.wa { color: #da4453; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>







<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">NRCRI_2021GS</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Imputation (DCas21_5841)
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="convertDCas21_5841_ToVCF.html">Conversion DArT report DCas21_5841 to VCF</a>
    </li>
    <li>
      <a href="ImputeDCas21_5841.html">Impute DCas21_5841</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Prediction (April 2021)
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="01-cleanTPdata.html">Data Cleaning</a>
    </li>
    <li>
      <a href="02-GetBLUPs.html">Get BLUPs</a>
    </li>
    <li>
      <a href="03-CrossValidation.html">Check prediction accuracy</a>
    </li>
    <li>
      <a href="04-GetGBLUPs.html">Get GEBV/GETGV for GS C3b</a>
    </li>
  </ul>
</li>
<li>
  <a href="05-Results.html">Results</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Other
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="about.html">About</a>
    </li>
    <li>
      <a href="license.html">License</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/wolfemd/NRCRI_2021GS">
    <span class="fas fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">NRCRI 2021 GS and related procedures</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-05-05
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>NRCRI_2021GS/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20210427code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20210427)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20210427code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20210427)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomwolfemdNRCRI2021GStree94ec81122c0b7a209fbf75bf47ca3e2101a0f529targetblank94ec811a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/wolfemd/NRCRI_2021GS/tree/94ec81122c0b7a209fbf75bf47ca3e2101a0f529" target="_blank">94ec811</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomwolfemdNRCRI2021GStree94ec81122c0b7a209fbf75bf47ca3e2101a0f529targetblank94ec811a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/wolfemd/NRCRI_2021GS/tree/94ec81122c0b7a209fbf75bf47ca3e2101a0f529" target="_blank">94ec811</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    data/.DS_Store

Untracked files:
    Untracked:  code/convertDart2vcf.R
    Untracked:  code/gsFunctions.R
    Untracked:  code/imputationFunctions.R
    Untracked:  data/DatabaseDownload_2021April29/
    Untracked:  data/GBSdataMasterList_31818.csv
    Untracked:  data/IITA_GBStoPhenoMaster_33018.csv
    Untracked:  data/NRCRI_GBStoPhenoMaster_40318.csv
    Untracked:  data/Report-DCas21-5841/
    Untracked:  data/chr1_RefPanelAndGSprogeny_ReadyForGP_72719.fam
    Untracked:  output/BeagleLogs/
    Untracked:  output/DosageMatrix_DCas21_5841_WA_REFimputedAndFiltered.rds
    Untracked:  output/DosageMatrix_NRCRI_2021May03.rds
    Untracked:  output/GEBV_NRCRI_ModelA_2021May03.csv
    Untracked:  output/GETGV_NRCRI_ModelADE_2021May03.csv
    Untracked:  output/Kinship_AD_NRCRI_2021May03.rds
    Untracked:  output/Kinship_A_NRCRI_2021May03.rds
    Untracked:  output/Kinship_D_NRCRI_2021May03.rds
    Untracked:  output/NRCRI_CleanedTrialData_2021May03.rds
    Untracked:  output/NRCRI_ExptDesignsDetected_2021May03.rds
    Untracked:  output/NRCRI_blupsForModelTraining_twostage_asreml_2021May03.rds
    Untracked:  output/NRCRI_trials_NOT_identifiable.csv
    Untracked:  output/OnlyChosen_germplasmName_to_FullSampleName_matches_NRCRI_2021May03.csv
    Untracked:  output/chr10_DCas21_5841_WA_REFimputed.INFO
    Untracked:  output/chr10_DCas21_5841_WA_REFimputed.hwe
    Untracked:  output/chr10_DCas21_5841_WA_REFimputed.log
    Untracked:  output/chr10_DCas21_5841_WA_REFimputed.sitesPassing
    Untracked:  output/chr10_DCas21_5841_WA_REFimputed.vcf.gz
    Untracked:  output/chr10_DCas21_5841_WA_REFimputedAndFiltered.alleleToCount
    Untracked:  output/chr10_DCas21_5841_WA_REFimputedAndFiltered.bed
    Untracked:  output/chr10_DCas21_5841_WA_REFimputedAndFiltered.bim
    Untracked:  output/chr10_DCas21_5841_WA_REFimputedAndFiltered.fam
    Untracked:  output/chr10_DCas21_5841_WA_REFimputedAndFiltered.log
    Untracked:  output/chr10_DCas21_5841_WA_REFimputedAndFiltered.nosex
    Untracked:  output/chr10_DCas21_5841_WA_REFimputedAndFiltered.raw
    Untracked:  output/chr10_DCas21_5841_WA_REFimputedAndFiltered.sitesWithAlleles
    Untracked:  output/chr10_DCas21_5841_WA_REFimputedAndFiltered.vcf.gz
    Untracked:  output/chr11_DCas21_5841_WA_REFimputed.INFO
    Untracked:  output/chr11_DCas21_5841_WA_REFimputed.hwe
    Untracked:  output/chr11_DCas21_5841_WA_REFimputed.log
    Untracked:  output/chr11_DCas21_5841_WA_REFimputed.sitesPassing
    Untracked:  output/chr11_DCas21_5841_WA_REFimputed.vcf.gz
    Untracked:  output/chr11_DCas21_5841_WA_REFimputedAndFiltered.alleleToCount
    Untracked:  output/chr11_DCas21_5841_WA_REFimputedAndFiltered.bed
    Untracked:  output/chr11_DCas21_5841_WA_REFimputedAndFiltered.bim
    Untracked:  output/chr11_DCas21_5841_WA_REFimputedAndFiltered.fam
    Untracked:  output/chr11_DCas21_5841_WA_REFimputedAndFiltered.log
    Untracked:  output/chr11_DCas21_5841_WA_REFimputedAndFiltered.nosex
    Untracked:  output/chr11_DCas21_5841_WA_REFimputedAndFiltered.raw
    Untracked:  output/chr11_DCas21_5841_WA_REFimputedAndFiltered.sitesWithAlleles
    Untracked:  output/chr11_DCas21_5841_WA_REFimputedAndFiltered.vcf.gz
    Untracked:  output/chr12_DCas21_5841_WA_REFimputed.INFO
    Untracked:  output/chr12_DCas21_5841_WA_REFimputed.hwe
    Untracked:  output/chr12_DCas21_5841_WA_REFimputed.log
    Untracked:  output/chr12_DCas21_5841_WA_REFimputed.sitesPassing
    Untracked:  output/chr12_DCas21_5841_WA_REFimputed.vcf.gz
    Untracked:  output/chr12_DCas21_5841_WA_REFimputedAndFiltered.alleleToCount
    Untracked:  output/chr12_DCas21_5841_WA_REFimputedAndFiltered.bed
    Untracked:  output/chr12_DCas21_5841_WA_REFimputedAndFiltered.bim
    Untracked:  output/chr12_DCas21_5841_WA_REFimputedAndFiltered.fam
    Untracked:  output/chr12_DCas21_5841_WA_REFimputedAndFiltered.log
    Untracked:  output/chr12_DCas21_5841_WA_REFimputedAndFiltered.nosex
    Untracked:  output/chr12_DCas21_5841_WA_REFimputedAndFiltered.raw
    Untracked:  output/chr12_DCas21_5841_WA_REFimputedAndFiltered.sitesWithAlleles
    Untracked:  output/chr12_DCas21_5841_WA_REFimputedAndFiltered.vcf.gz
    Untracked:  output/chr13_DCas21_5841_WA_REFimputed.INFO
    Untracked:  output/chr13_DCas21_5841_WA_REFimputed.hwe
    Untracked:  output/chr13_DCas21_5841_WA_REFimputed.log
    Untracked:  output/chr13_DCas21_5841_WA_REFimputed.sitesPassing
    Untracked:  output/chr13_DCas21_5841_WA_REFimputed.vcf.gz
    Untracked:  output/chr13_DCas21_5841_WA_REFimputedAndFiltered.alleleToCount
    Untracked:  output/chr13_DCas21_5841_WA_REFimputedAndFiltered.bed
    Untracked:  output/chr13_DCas21_5841_WA_REFimputedAndFiltered.bim
    Untracked:  output/chr13_DCas21_5841_WA_REFimputedAndFiltered.fam
    Untracked:  output/chr13_DCas21_5841_WA_REFimputedAndFiltered.log
    Untracked:  output/chr13_DCas21_5841_WA_REFimputedAndFiltered.nosex
    Untracked:  output/chr13_DCas21_5841_WA_REFimputedAndFiltered.raw
    Untracked:  output/chr13_DCas21_5841_WA_REFimputedAndFiltered.sitesWithAlleles
    Untracked:  output/chr13_DCas21_5841_WA_REFimputedAndFiltered.vcf.gz
    Untracked:  output/chr14_DCas21_5841_WA_REFimputed.INFO
    Untracked:  output/chr14_DCas21_5841_WA_REFimputed.hwe
    Untracked:  output/chr14_DCas21_5841_WA_REFimputed.log
    Untracked:  output/chr14_DCas21_5841_WA_REFimputed.sitesPassing
    Untracked:  output/chr14_DCas21_5841_WA_REFimputed.vcf.gz
    Untracked:  output/chr14_DCas21_5841_WA_REFimputedAndFiltered.alleleToCount
    Untracked:  output/chr14_DCas21_5841_WA_REFimputedAndFiltered.bed
    Untracked:  output/chr14_DCas21_5841_WA_REFimputedAndFiltered.bim
    Untracked:  output/chr14_DCas21_5841_WA_REFimputedAndFiltered.fam
    Untracked:  output/chr14_DCas21_5841_WA_REFimputedAndFiltered.log
    Untracked:  output/chr14_DCas21_5841_WA_REFimputedAndFiltered.nosex
    Untracked:  output/chr14_DCas21_5841_WA_REFimputedAndFiltered.raw
    Untracked:  output/chr14_DCas21_5841_WA_REFimputedAndFiltered.sitesWithAlleles
    Untracked:  output/chr14_DCas21_5841_WA_REFimputedAndFiltered.vcf.gz
    Untracked:  output/chr15_DCas21_5841_WA_REFimputed.INFO
    Untracked:  output/chr15_DCas21_5841_WA_REFimputed.hwe
    Untracked:  output/chr15_DCas21_5841_WA_REFimputed.log
    Untracked:  output/chr15_DCas21_5841_WA_REFimputed.sitesPassing
    Untracked:  output/chr15_DCas21_5841_WA_REFimputed.vcf.gz
    Untracked:  output/chr15_DCas21_5841_WA_REFimputedAndFiltered.alleleToCount
    Untracked:  output/chr15_DCas21_5841_WA_REFimputedAndFiltered.bed
    Untracked:  output/chr15_DCas21_5841_WA_REFimputedAndFiltered.bim
    Untracked:  output/chr15_DCas21_5841_WA_REFimputedAndFiltered.fam
    Untracked:  output/chr15_DCas21_5841_WA_REFimputedAndFiltered.log
    Untracked:  output/chr15_DCas21_5841_WA_REFimputedAndFiltered.nosex
    Untracked:  output/chr15_DCas21_5841_WA_REFimputedAndFiltered.raw
    Untracked:  output/chr15_DCas21_5841_WA_REFimputedAndFiltered.sitesWithAlleles
    Untracked:  output/chr15_DCas21_5841_WA_REFimputedAndFiltered.vcf.gz
    Untracked:  output/chr16_DCas21_5841_WA_REFimputed.INFO
    Untracked:  output/chr16_DCas21_5841_WA_REFimputed.hwe
    Untracked:  output/chr16_DCas21_5841_WA_REFimputed.log
    Untracked:  output/chr16_DCas21_5841_WA_REFimputed.sitesPassing
    Untracked:  output/chr16_DCas21_5841_WA_REFimputed.vcf.gz
    Untracked:  output/chr16_DCas21_5841_WA_REFimputedAndFiltered.alleleToCount
    Untracked:  output/chr16_DCas21_5841_WA_REFimputedAndFiltered.bed
    Untracked:  output/chr16_DCas21_5841_WA_REFimputedAndFiltered.bim
    Untracked:  output/chr16_DCas21_5841_WA_REFimputedAndFiltered.fam
    Untracked:  output/chr16_DCas21_5841_WA_REFimputedAndFiltered.log
    Untracked:  output/chr16_DCas21_5841_WA_REFimputedAndFiltered.nosex
    Untracked:  output/chr16_DCas21_5841_WA_REFimputedAndFiltered.raw
    Untracked:  output/chr16_DCas21_5841_WA_REFimputedAndFiltered.sitesWithAlleles
    Untracked:  output/chr16_DCas21_5841_WA_REFimputedAndFiltered.vcf.gz
    Untracked:  output/chr17_DCas21_5841_WA_REFimputed.INFO
    Untracked:  output/chr17_DCas21_5841_WA_REFimputed.hwe
    Untracked:  output/chr17_DCas21_5841_WA_REFimputed.log
    Untracked:  output/chr17_DCas21_5841_WA_REFimputed.sitesPassing
    Untracked:  output/chr17_DCas21_5841_WA_REFimputed.vcf.gz
    Untracked:  output/chr17_DCas21_5841_WA_REFimputedAndFiltered.alleleToCount
    Untracked:  output/chr17_DCas21_5841_WA_REFimputedAndFiltered.bed
    Untracked:  output/chr17_DCas21_5841_WA_REFimputedAndFiltered.bim
    Untracked:  output/chr17_DCas21_5841_WA_REFimputedAndFiltered.fam
    Untracked:  output/chr17_DCas21_5841_WA_REFimputedAndFiltered.log
    Untracked:  output/chr17_DCas21_5841_WA_REFimputedAndFiltered.nosex
    Untracked:  output/chr17_DCas21_5841_WA_REFimputedAndFiltered.raw
    Untracked:  output/chr17_DCas21_5841_WA_REFimputedAndFiltered.sitesWithAlleles
    Untracked:  output/chr17_DCas21_5841_WA_REFimputedAndFiltered.vcf.gz
    Untracked:  output/chr18_DCas21_5841_WA_REFimputed.INFO
    Untracked:  output/chr18_DCas21_5841_WA_REFimputed.hwe
    Untracked:  output/chr18_DCas21_5841_WA_REFimputed.log
    Untracked:  output/chr18_DCas21_5841_WA_REFimputed.sitesPassing
    Untracked:  output/chr18_DCas21_5841_WA_REFimputed.vcf.gz
    Untracked:  output/chr18_DCas21_5841_WA_REFimputedAndFiltered.alleleToCount
    Untracked:  output/chr18_DCas21_5841_WA_REFimputedAndFiltered.bed
    Untracked:  output/chr18_DCas21_5841_WA_REFimputedAndFiltered.bim
    Untracked:  output/chr18_DCas21_5841_WA_REFimputedAndFiltered.fam
    Untracked:  output/chr18_DCas21_5841_WA_REFimputedAndFiltered.log
    Untracked:  output/chr18_DCas21_5841_WA_REFimputedAndFiltered.nosex
    Untracked:  output/chr18_DCas21_5841_WA_REFimputedAndFiltered.raw
    Untracked:  output/chr18_DCas21_5841_WA_REFimputedAndFiltered.sitesWithAlleles
    Untracked:  output/chr18_DCas21_5841_WA_REFimputedAndFiltered.vcf.gz
    Untracked:  output/chr1_DCas21_5841_WA_REFimputed.INFO
    Untracked:  output/chr1_DCas21_5841_WA_REFimputed.hwe
    Untracked:  output/chr1_DCas21_5841_WA_REFimputed.log
    Untracked:  output/chr1_DCas21_5841_WA_REFimputed.sitesPassing
    Untracked:  output/chr1_DCas21_5841_WA_REFimputed.vcf.gz
    Untracked:  output/chr1_DCas21_5841_WA_REFimputedAndFiltered.alleleToCount
    Untracked:  output/chr1_DCas21_5841_WA_REFimputedAndFiltered.bed
    Untracked:  output/chr1_DCas21_5841_WA_REFimputedAndFiltered.bim
    Untracked:  output/chr1_DCas21_5841_WA_REFimputedAndFiltered.fam
    Untracked:  output/chr1_DCas21_5841_WA_REFimputedAndFiltered.log
    Untracked:  output/chr1_DCas21_5841_WA_REFimputedAndFiltered.nosex
    Untracked:  output/chr1_DCas21_5841_WA_REFimputedAndFiltered.raw
    Untracked:  output/chr1_DCas21_5841_WA_REFimputedAndFiltered.sitesWithAlleles
    Untracked:  output/chr1_DCas21_5841_WA_REFimputedAndFiltered.vcf.gz
    Untracked:  output/chr2_DCas21_5841_WA_REFimputed.INFO
    Untracked:  output/chr2_DCas21_5841_WA_REFimputed.hwe
    Untracked:  output/chr2_DCas21_5841_WA_REFimputed.log
    Untracked:  output/chr2_DCas21_5841_WA_REFimputed.sitesPassing
    Untracked:  output/chr2_DCas21_5841_WA_REFimputed.vcf.gz
    Untracked:  output/chr2_DCas21_5841_WA_REFimputedAndFiltered.alleleToCount
    Untracked:  output/chr2_DCas21_5841_WA_REFimputedAndFiltered.bed
    Untracked:  output/chr2_DCas21_5841_WA_REFimputedAndFiltered.bim
    Untracked:  output/chr2_DCas21_5841_WA_REFimputedAndFiltered.fam
    Untracked:  output/chr2_DCas21_5841_WA_REFimputedAndFiltered.log
    Untracked:  output/chr2_DCas21_5841_WA_REFimputedAndFiltered.nosex
    Untracked:  output/chr2_DCas21_5841_WA_REFimputedAndFiltered.raw
    Untracked:  output/chr2_DCas21_5841_WA_REFimputedAndFiltered.sitesWithAlleles
    Untracked:  output/chr2_DCas21_5841_WA_REFimputedAndFiltered.vcf.gz
    Untracked:  output/chr3_DCas21_5841_WA_REFimputed.INFO
    Untracked:  output/chr3_DCas21_5841_WA_REFimputed.hwe
    Untracked:  output/chr3_DCas21_5841_WA_REFimputed.log
    Untracked:  output/chr3_DCas21_5841_WA_REFimputed.sitesPassing
    Untracked:  output/chr3_DCas21_5841_WA_REFimputed.vcf.gz
    Untracked:  output/chr3_DCas21_5841_WA_REFimputedAndFiltered.alleleToCount
    Untracked:  output/chr3_DCas21_5841_WA_REFimputedAndFiltered.bed
    Untracked:  output/chr3_DCas21_5841_WA_REFimputedAndFiltered.bim
    Untracked:  output/chr3_DCas21_5841_WA_REFimputedAndFiltered.fam
    Untracked:  output/chr3_DCas21_5841_WA_REFimputedAndFiltered.log
    Untracked:  output/chr3_DCas21_5841_WA_REFimputedAndFiltered.nosex
    Untracked:  output/chr3_DCas21_5841_WA_REFimputedAndFiltered.raw
    Untracked:  output/chr3_DCas21_5841_WA_REFimputedAndFiltered.sitesWithAlleles
    Untracked:  output/chr3_DCas21_5841_WA_REFimputedAndFiltered.vcf.gz
    Untracked:  output/chr4_DCas21_5841_WA_REFimputed.INFO
    Untracked:  output/chr4_DCas21_5841_WA_REFimputed.hwe
    Untracked:  output/chr4_DCas21_5841_WA_REFimputed.log
    Untracked:  output/chr4_DCas21_5841_WA_REFimputed.sitesPassing
    Untracked:  output/chr4_DCas21_5841_WA_REFimputed.vcf.gz
    Untracked:  output/chr4_DCas21_5841_WA_REFimputedAndFiltered.alleleToCount
    Untracked:  output/chr4_DCas21_5841_WA_REFimputedAndFiltered.bed
    Untracked:  output/chr4_DCas21_5841_WA_REFimputedAndFiltered.bim
    Untracked:  output/chr4_DCas21_5841_WA_REFimputedAndFiltered.fam
    Untracked:  output/chr4_DCas21_5841_WA_REFimputedAndFiltered.log
    Untracked:  output/chr4_DCas21_5841_WA_REFimputedAndFiltered.nosex
    Untracked:  output/chr4_DCas21_5841_WA_REFimputedAndFiltered.raw
    Untracked:  output/chr4_DCas21_5841_WA_REFimputedAndFiltered.sitesWithAlleles
    Untracked:  output/chr4_DCas21_5841_WA_REFimputedAndFiltered.vcf.gz
    Untracked:  output/chr5_DCas21_5841_WA_REFimputed.INFO
    Untracked:  output/chr5_DCas21_5841_WA_REFimputed.hwe
    Untracked:  output/chr5_DCas21_5841_WA_REFimputed.log
    Untracked:  output/chr5_DCas21_5841_WA_REFimputed.sitesPassing
    Untracked:  output/chr5_DCas21_5841_WA_REFimputed.vcf.gz
    Untracked:  output/chr5_DCas21_5841_WA_REFimputedAndFiltered.alleleToCount
    Untracked:  output/chr5_DCas21_5841_WA_REFimputedAndFiltered.bed
    Untracked:  output/chr5_DCas21_5841_WA_REFimputedAndFiltered.bim
    Untracked:  output/chr5_DCas21_5841_WA_REFimputedAndFiltered.fam
    Untracked:  output/chr5_DCas21_5841_WA_REFimputedAndFiltered.log
    Untracked:  output/chr5_DCas21_5841_WA_REFimputedAndFiltered.nosex
    Untracked:  output/chr5_DCas21_5841_WA_REFimputedAndFiltered.raw
    Untracked:  output/chr5_DCas21_5841_WA_REFimputedAndFiltered.sitesWithAlleles
    Untracked:  output/chr5_DCas21_5841_WA_REFimputedAndFiltered.vcf.gz
    Untracked:  output/chr6_DCas21_5841_WA_REFimputed.INFO
    Untracked:  output/chr6_DCas21_5841_WA_REFimputed.hwe
    Untracked:  output/chr6_DCas21_5841_WA_REFimputed.log
    Untracked:  output/chr6_DCas21_5841_WA_REFimputed.sitesPassing
    Untracked:  output/chr6_DCas21_5841_WA_REFimputed.vcf.gz
    Untracked:  output/chr6_DCas21_5841_WA_REFimputedAndFiltered.alleleToCount
    Untracked:  output/chr6_DCas21_5841_WA_REFimputedAndFiltered.bed
    Untracked:  output/chr6_DCas21_5841_WA_REFimputedAndFiltered.bim
    Untracked:  output/chr6_DCas21_5841_WA_REFimputedAndFiltered.fam
    Untracked:  output/chr6_DCas21_5841_WA_REFimputedAndFiltered.log
    Untracked:  output/chr6_DCas21_5841_WA_REFimputedAndFiltered.nosex
    Untracked:  output/chr6_DCas21_5841_WA_REFimputedAndFiltered.raw
    Untracked:  output/chr6_DCas21_5841_WA_REFimputedAndFiltered.sitesWithAlleles
    Untracked:  output/chr6_DCas21_5841_WA_REFimputedAndFiltered.vcf.gz
    Untracked:  output/chr7_DCas21_5841_WA_REFimputed.INFO
    Untracked:  output/chr7_DCas21_5841_WA_REFimputed.hwe
    Untracked:  output/chr7_DCas21_5841_WA_REFimputed.log
    Untracked:  output/chr7_DCas21_5841_WA_REFimputed.sitesPassing
    Untracked:  output/chr7_DCas21_5841_WA_REFimputed.vcf.gz
    Untracked:  output/chr7_DCas21_5841_WA_REFimputedAndFiltered.alleleToCount
    Untracked:  output/chr7_DCas21_5841_WA_REFimputedAndFiltered.bed
    Untracked:  output/chr7_DCas21_5841_WA_REFimputedAndFiltered.bim
    Untracked:  output/chr7_DCas21_5841_WA_REFimputedAndFiltered.fam
    Untracked:  output/chr7_DCas21_5841_WA_REFimputedAndFiltered.log
    Untracked:  output/chr7_DCas21_5841_WA_REFimputedAndFiltered.nosex
    Untracked:  output/chr7_DCas21_5841_WA_REFimputedAndFiltered.raw
    Untracked:  output/chr7_DCas21_5841_WA_REFimputedAndFiltered.sitesWithAlleles
    Untracked:  output/chr7_DCas21_5841_WA_REFimputedAndFiltered.vcf.gz
    Untracked:  output/chr8_DCas21_5841_WA_REFimputed.INFO
    Untracked:  output/chr8_DCas21_5841_WA_REFimputed.hwe
    Untracked:  output/chr8_DCas21_5841_WA_REFimputed.log
    Untracked:  output/chr8_DCas21_5841_WA_REFimputed.sitesPassing
    Untracked:  output/chr8_DCas21_5841_WA_REFimputed.vcf.gz
    Untracked:  output/chr8_DCas21_5841_WA_REFimputedAndFiltered.alleleToCount
    Untracked:  output/chr8_DCas21_5841_WA_REFimputedAndFiltered.bed
    Untracked:  output/chr8_DCas21_5841_WA_REFimputedAndFiltered.bim
    Untracked:  output/chr8_DCas21_5841_WA_REFimputedAndFiltered.fam
    Untracked:  output/chr8_DCas21_5841_WA_REFimputedAndFiltered.log
    Untracked:  output/chr8_DCas21_5841_WA_REFimputedAndFiltered.nosex
    Untracked:  output/chr8_DCas21_5841_WA_REFimputedAndFiltered.raw
    Untracked:  output/chr8_DCas21_5841_WA_REFimputedAndFiltered.sitesWithAlleles
    Untracked:  output/chr8_DCas21_5841_WA_REFimputedAndFiltered.vcf.gz
    Untracked:  output/chr9_DCas21_5841_WA_REFimputed.INFO
    Untracked:  output/chr9_DCas21_5841_WA_REFimputed.hwe
    Untracked:  output/chr9_DCas21_5841_WA_REFimputed.log
    Untracked:  output/chr9_DCas21_5841_WA_REFimputed.sitesPassing
    Untracked:  output/chr9_DCas21_5841_WA_REFimputed.vcf.gz
    Untracked:  output/chr9_DCas21_5841_WA_REFimputedAndFiltered.alleleToCount
    Untracked:  output/chr9_DCas21_5841_WA_REFimputedAndFiltered.bed
    Untracked:  output/chr9_DCas21_5841_WA_REFimputedAndFiltered.bim
    Untracked:  output/chr9_DCas21_5841_WA_REFimputedAndFiltered.fam
    Untracked:  output/chr9_DCas21_5841_WA_REFimputedAndFiltered.log
    Untracked:  output/chr9_DCas21_5841_WA_REFimputedAndFiltered.nosex
    Untracked:  output/chr9_DCas21_5841_WA_REFimputedAndFiltered.raw
    Untracked:  output/chr9_DCas21_5841_WA_REFimputedAndFiltered.sitesWithAlleles
    Untracked:  output/chr9_DCas21_5841_WA_REFimputedAndFiltered.vcf.gz
    Untracked:  output/cvresults_ADE_1_2021May03.rds
    Untracked:  output/cvresults_ADE_2_2021May03.rds
    Untracked:  output/cvresults_A_2021May03.rds
    Untracked:  output/genomicPredictions_ModelADE_twostage_NRCRI_2021May03.rds
    Untracked:  output/genomicPredictions_ModelA_twostage_NRCRI_2021May03.rds
    Untracked:  output/genomicPredictions_NRCRI_2021May03.csv
    Untracked:  output/maxNOHAV_byStudy.csv

Unstaged changes:
    Modified:   README.md

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/index.Rmd</code>) and HTML (<code>docs/index.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/wolfemd/NRCRI_2021GS/blob/94ec81122c0b7a209fbf75bf47ca3e2101a0f529/analysis/index.Rmd" target="_blank">94ec811</a>
</td>
<td>
wolfemd
</td>
<td>
2021-05-05
</td>
<td>
Completed prediction. Build site and push to GitHub / Cassavabase FTP server. Share to NRCRI.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/wolfemd/NRCRI_2021GS/e8b3eebcbb3c59166c3122f5960291e79345a1c6/docs/index.html" target="_blank">e8b3eeb</a>
</td>
<td>
wolfemd
</td>
<td>
2021-05-03
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/wolfemd/NRCRI_2021GS/blob/c8876397f11b6301a3caa17a2a6daeed125ef27b/analysis/index.Rmd" target="_blank">c887639</a>
</td>
<td>
wolfemd
</td>
<td>
2021-05-03
</td>
<td>
Publish site up through cleanTPdata step to generate cleaned TP data before continuing pipeline.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/wolfemd/NRCRI_2021GS/blob/46276078dc4ea3360132384828c4f0ee6f1b04e3/analysis/index.Rmd" target="_blank">4627607</a>
</td>
<td>
wolfemd
</td>
<td>
2021-04-27
</td>
<td>
Start workflowr project.
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>Welcome to my research website.</p>
<p>This repository and website documents all analyses, summary, tables and figures associated with NRCRI genomic prediction and related procedures (e.g. imputation).</p>
<div id="april-2021-dcas21_5841---nrcri-c3b" class="section level1">
<h1>April 2021 (DCas21_5841 - NRCRI C3b)</h1>
<p>I think this report corresponds to the GS C3b. Last NRCRI imputation and prediction was <a href="https://wolfemd.github.io/NRCRI_2020GS/index.html#october-imputations">October 2020</a> corresponding to a C3a, I assume.</p>
<div id="imputation-of-genotyping-project" class="section level2">
<h2>Imputation of genotyping project</h2>
<p>Impute with E. Africa Imputation Reference Panel dataset, which can be found on the <a href="ftp://ftp.cassavabase.org/marnin_datasets/nextgenImputation2019/ImputationEastAfrica_StageII_90919/">Cassavabase FTP server here</a> with names e.g. <code>chr*_ImputationReferencePanel_StageVI_91119.vcf.gz</code> with <a href="https://wolfemd.github.io/NaCRRI_2020GS/">code/documentation here</a>.</p>
<p><strong>Steps</strong>:</p>
<ol style="list-style-type: decimal">
<li><a href="convertDCas21_5841_ToVCF.html">Convert DCas21-5841 report to VCF for imputation</a>:</li>
<li><a href="ImputeDCas21_5841.html">Impute DCas21-5841</a>: with West Africa reference panel</li>
</ol>
<p><strong>Files</strong>:</p>
<ul>
<li><strong>RefPanel VCF filename:</strong> <code>chr*_ImputationReferencePanel_StageIIpartI_72219.vcf.gz</code></li>
<li><strong>Imputed filename:</strong> <code>chr*_DCas21_5841_WA_REFimputed.vcf.gz</code></li>
<li><strong>Post-impute filtered filename:</strong> <code>chr*_DCas21_5841_WA_REFimputedAndFiltered.vcf.gz</code></li>
<li><strong>Genome-wide dosage matrix format for use in R:</strong> <code>DosageMatrix_DCas21_5841_WA_REFimputedAndFiltered.rds</code></li>
</ul>
<p><strong>HOW TO COMBINE DOSAGE MATRICES:</strong> Users will want to combine the genotypes in the imputation reference panel files, with the genotypes in the imputed DArT file. They can have slightly different sets of markers along the columns. Here is a basic example how to combine:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a>snps_refpanel<span class="ot">&lt;-</span><span class="fu">readRDS</span>(<span class="st">&quot;DosageMatrix_RefPanelAndGSprogeny_ReadyForGP_73019.rds&quot;</span>)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a>snps_dcas21_5841<span class="ot">&lt;-</span><span class="fu">readRDS</span>(<span class="st">&quot;DosageMatrix_DCas21_5841_WA_REFimputedAndFiltered.rds&quot;</span>)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a>snps2keep<span class="ot">&lt;-</span><span class="fu">colnames</span>(snps_refpanel)[,<span class="fu">colnames</span>(snps_refpanel) <span class="sc">%in%</span> <span class="fu">colnames</span>(snps_dcas20_5629)]</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a>snps<span class="ot">&lt;-</span><span class="fu">bind_rows</span>(snps_refpanel[,snps2keep],</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a>                snps_dcas21_5841[,snps2keep])</span></code></pre></div>
</div>
<div id="genomic-prediction" class="section level2">
<h2>Genomic Prediction</h2>
<p>Get NRCRI TP data from Cassavabase. Use it with imputed data to predict GEBV/GETGV for all samples in the new reports (<strong>DCas21-5841</strong>).</p>
<ol style="list-style-type: decimal">
<li><a href="01-cleanTPdata.html">Prepare training dataset</a>: Download data from DB, “Clean” and format DB data.</li>
<li><a href="02-GetBLUPs.html">Get BLUPs combining all trial data</a>: Combine data from all trait-trials to get BLUPs for downstream genomic prediction.
<ul>
<li>Fit mixed-model to multi-trial dataset and extract BLUPs, de-regressed BLUPs and weights. Include two rounds of outlier removal.</li>
</ul></li>
<li><a href="03-CrossValidation.html">Check prediction accuracy</a>: Evaluate prediction accuracy with cross-validation.
<ul>
<li>Compare prediction accuracy with vs. without IITA’s training data to augment.</li>
</ul></li>
<li><a href="04-GetGBLUPs.html">Genomic prediction</a>: Predict <em>genomic</em> BLUPs (GEBV and GETGV) for all selection candidates using all available data.</li>
<li><a href="05-Results.html">Results</a>: Plots, results and recommendations.</li>
</ol>
<p><strong>OUTPUT / FILES</strong>: everything is in the <code>output/</code> sub-directory.</p>
<ul>
<li><strong>GEBVs for parent selection:</strong> <code>GEBV_NRCRI_ModelA_2021May03.csv</code></li>
<li><strong>GETGVs for variety advancement:</strong> <code>GETGV_NRCRI_ModelADE_2021May03.csv</code></li>
<li><strong>Tidy, long-form CSV of predictions, including PEVs:</strong> <code>genomicPredictions_NRCRI_2021May03.csv</code></li>
</ul>
<p><a href="ftp://ftp.cassavabase.org/marnin_datasets/NRCRI_2021GS/output/"><strong>DOWNLOAD FROM CASSAVABASE FTP SERVER</strong></a></p>
<p>or</p>
<p><a href="https://github.com/wolfemd/NRCRI_2021GS/tree/master/output"><strong>DOWNLOAD FROM GitHub</strong></a></p>
</div>
</div>
<div id="data-availability-and-reproducibility" class="section level1">
<h1>Data availability and reproducibility</h1>
<p>The R package <strong>workflowr</strong> was used to document this study reproducibly.</p>
<p>Much of the supporting data <em>and</em> output from the analyses documented here are too large for GitHub.</p>
<p>The repository will be mirrored, here: <a href="ftp://ftp.cassavabase.org/marnin_datasets/NRCRI_2021GS/" class="uri">ftp://ftp.cassavabase.org/marnin_datasets/NRCRI_2021GS/</a> with all data.</p>
</div>
<div id="directory-structure-of-this-repository" class="section level1">
<h1>Directory structure of this repository</h1>
<p><strong>NOTICE:</strong> <code>data/</code> and <code>output/</code> are empty on GitHub. Please see <a href="ftp://ftp.cassavabase.org/marnin_datasets/NRCRI_2021GS/" class="uri">ftp://ftp.cassavabase.org/marnin_datasets/NRCRI_2021GS/</a> for access.</p>
<ol style="list-style-type: decimal">
<li><code>data/</code>: raw data (e.g. unimputed SNP data)</li>
<li><code>output/</code>: outputs (e.g. imputed SNP data)</li>
<li><code>analysis/</code>: most code and workflow documented in <strong>.Rmd</strong> files</li>
<li><code>docs/</code>: compiled <strong>.html</strong>, “knitted” from <strong>.Rmd</strong></li>
</ol>
<p>Supporting functions <code>code/</code></p>
<p>The analyses in the <strong>html</strong> / <strong>Rmd</strong> files referenced above often source R scripts in the <code>code/</code> sub-folder.</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code></pre></div>
<pre><code>R version 4.0.3 (2020-10-10)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] workflowr_1.6.2

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.6        whisker_0.4       knitr_1.32        magrittr_2.0.1   
 [5] R6_2.5.0          rlang_0.4.10      fansi_0.4.2       stringr_1.4.0    
 [9] tools_4.0.3       xfun_0.22         utf8_1.2.1        git2r_0.28.0     
[13] jquerylib_0.1.3   htmltools_0.5.1.1 ellipsis_0.3.1    rprojroot_2.0.2  
[17] yaml_2.2.1        digest_0.6.27     tibble_3.1.1      lifecycle_1.0.0  
[21] crayon_1.4.1      later_1.1.0.1     sass_0.3.1        vctrs_0.3.7      
[25] promises_1.2.0.1  fs_1.5.0          glue_1.4.2        evaluate_0.14    
[29] rmarkdown_2.7     stringi_1.5.3     bslib_0.2.4       compiler_4.0.3   
[33] pillar_1.6.0      jsonlite_1.7.2    httpuv_1.5.5      pkgconfig_2.0.3  </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>





</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
